pkg/controllers/cloud/networkpolicy.go

```diff
@@ -438,6 +439,14 @@ func (s *securityGroupImpl) deleteImpl(c cloudSecurityGroup, membershipOnly bool
 	ch := securitygroup.CloudSecurityGroup.DeleteSecurityGroup(&s.id, membershipOnly)
 	go func() {
 		err := <-ch
+		if err == nil && !membershipOnly {
+			// remove sg rules from the cache.
+			rules, e := r.cloudRuleIndexer.ByIndex(cloudRuleIndexerByAppliedToGrp, s.id.CloudResourceID.String())
+			if e != nil {
+				r.Log.Error(e, "get cloudRule indexer", "index", s.id.CloudResourceID.String())
+			}
+			_ = utils.DeleteIndexerObjects(r.cloudRuleIndexer, rules)
```

Comment:
wont rules be nil, if error occurs ?

Reply:
Yes, but nil slice is still slice, the function still works. We can add a check for nil to optimize a bit though.